packages/x-date-pickers/src/PickersDay/PickersDay.tsx

```diff
@@ -327,7 +327,7 @@ const PickersDayRaw = React.forwardRef(function PickersDay<TDate>(
       tabIndex={selected ? 0 : -1}
       onFocus={handleFocus}
       onKeyDown={handleKeyDown}
-      onClick={handleClick}
+      onMouseDown={handleMouseDown}
```

Comment:
We could also add a couple of tests to avoid breaking this behaviour in the future. 😉 
```tsx
it("should call `onChange` when selecting a date with `Enter` key", async () => {
  const onChange = spy();

  render(
    <WrappedDesktopDatePicker
      onChange={onChange}
      initialValue={adapterToUse.date(new Date(2018, 0, 1))}
    />
  );

  openPicker({ type: "date", variant: "desktop" });

  fireEvent.keyDown(screen.getByLabelText("Jan 1, 2018"), {
    key: "ArrowRight",
    code: "ArrowRight",
  });
  fireEvent.keyDown(screen.getByLabelText("Jan 2, 2018"), {
    key: "Enter",
    code: "Enter",
  });
  expect(onChange.callCount).to.equal(1);
  expect(onChange.lastCall.args[0]).toEqualDateTime(new Date(2018, 0, 2));
});

it("should call `onChange` when selecting a date with `Space` key", async () => {
  const onChange = spy();

  render(
    <WrappedDesktopDatePicker
      onChange={onChange}
      initialValue={adapterToUse.date(new Date(2018, 0, 5))}
    />
  );

  openPicker({ type: "date", variant: "desktop" });

  fireEvent.keyDown(screen.getByLabelText("Jan 5, 2018"), {
    key: "ArrowLeft",
    code: "ArrowLeft",
  });
  fireEvent.keyDown(screen.getByLabelText("Jan 4, 2018"), {
    key: "Enter",
    code: "Enter",
  });
  expect(onChange.callCount).to.equal(1);
  expect(onChange.lastCall.args[0]).toEqualDateTime(new Date(2018, 0, 4));
});

```

Reply:
That's a good catch. 

Could rename `handleMouseDown` to `handleClick` and have `onMouseDown={handleClick}`, and 
```jsx
case 'Enter':
case 'Space':
  // using `onMouseDown` instead of `onClick` no longer triggers the corresponding listener on `Enter` or `Space` click
  handleClick(event)
  break;
```

But yes would be better to have the `onClick`

----

Another solution could be to prevent default on `mouseDown` such that the focus is not applied (an so not month switch during the click) This would implies to manage the `handleFocus` in the `onClick` to stay consistent